executor.py

```diff
@@ -1,7 +1,130 @@
+            getattr(self.metas, 'name', self.__class__.__name__)
+        ).logger
+
     @requests
-    def foo(self, docs: DocumentArray, **kwargs):
-        pass
+    def encode(self, docs: DocumentArray, parameters: Dict = {}, **kwargs):
+        """
+        Read the numpy arrays from the Document.blob, and encode it into embeddings using the
+        comparable hashing technique. Features in the image are used to generate a distinct
+        (but not unique) fingerprint, and these fingerprints are comparable.
+
+        :param docs: A document array with documents to create embeddings for. Only the
+            documents that have the ``blob`` attribute will get embeddings. The ``blob``
+            attribute should be the numpy array of the image, and should have dtype
+            ``np.uint8``
+        :param parameters: A dictionary that contains parameters to control hash encoding.
+            The accepted keys are: `hash_type`, `hash_size`, `hash_func_args`, and `traversal_paths`.
+            For example: `parameters={'hash_type': 'phash', 'hash_func_args': {'highfreq_factor': 8}}`
+        """
+        missing_blob = []
+        hash_type = parameters.get('hash_type', self.hash_type)
+        hash_size = parameters.get('hash_size', self.hash_size)
+        hash_func_args = deepcopy(self._hash_func_args)
+        hash_func_args.update(parameters.get('hash_func_args', {}))
+
+        if not hash_type or hash_type not in HASH_TYPE:
+            raise ValueError(
+                f'Please select one of the available `hash_type`: {HASH_TYPE}'
+            )
+
+        for doc in docs.traverse_flat(
+            parameters.get('traversal_paths', self.traversal_paths)
+        ):
+            if doc.blob is None:
+                missing_blob.append(doc.id)
+                continue
+
+            image = Image.fromarray(doc.blob)
+            hash_hex = None
+            try:
+                if hash_type == 'whash':
+                    hash_hex = imagehash.whash(
+                        image, hash_size=hash_size, **hash_func_args
+                    )
+                elif hash_type == 'average_hash':
+                    hash_hex = imagehash.average_hash(
+                        image,
+                        hash_size=hash_size,
+                    )
+                elif hash_type == 'phash':
+                    hash_hex = imagehash.phash(
+                        image, hash_size=hash_size, **hash_func_args
+                    )
+                else:
+                    hash_hex = imagehash.dhash(image, hash_size=hash_size)
+            except AssertionError as e:
+                self.logger.error(f'Image hashing failed, {e}')
+
+            if hash_hex is not None:
+                if self.is_embed_bool:
+                    doc.embedding = np.squeeze(
+                        np.array(hash_hex.hash).reshape(1, hash_size ** 2)
+                    )
+                else:
+                    doc.embedding = np.squeeze(
```

Comment:
`doc.embedding = np.squeeze(` 
should be outside of the if block.
Instead assign the ndarray and have the call of squeeze once afterwards